src/test/java/GmailActionableEmailsServletTest.java

```diff
@@ -0,0 +1,179 @@
+// Copyright 2020 Google LLC
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     https://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+import com.google.api.services.gmail.model.Message;
+import com.google.api.services.gmail.model.MessagePart;
+import com.google.api.services.gmail.model.MessagePartHeader;
+import com.google.common.collect.ImmutableList;
+import com.google.gson.Gson;
+import com.google.gson.reflect.TypeToken;
+import com.google.sps.model.GmailClient;
+import com.google.sps.model.GmailClientFactory;
+import com.google.sps.servlets.GmailActionableEmailsServlet;
+import java.lang.reflect.Type;
+import java.util.Arrays;
+import java.util.Collections;
+import java.util.List;
+import org.junit.Assert;
+import org.junit.Before;
+import org.junit.Test;
+import org.junit.runner.RunWith;
+import org.junit.runners.JUnit4;
+import org.mockito.Mockito;
+
+/** Tests the GmailActionableEmailsServlet */
+@RunWith(JUnit4.class)
+public class GmailActionableEmailsServletTest extends AuthenticatedServletTestBase {
+  private GmailClient gmailClient;
+  private GmailActionableEmailsServlet servlet;
+
+  private static final Gson gson = new Gson();
+
+  // spaces should be parsed correctly by request.getParameter if query string is properly encoded
```

Comment:
very minor nit: Spaces instead of spaces

Reply:
Fixed! Thanks!

Reply:
`RESOLVED`